geoutils/geovector.py

```diff
@@ -158,8 +156,111 @@ def create_mask(self, rst=None, crs=None, xres=None, yres=None, bounds=None, in_
+        :type yres: float
+        :param bounds: Output raster bounds (left, bottom, right, top). Only if rst is None (Default to self bounds)
+        :type bounds: tuple
+        :param in_value: Value(s) to be burned inside the polygons (Default is self.ds.index + 1)
+        :type in_value: int, float, iterable
+        :param out_value: Value to be burned outside the polygons (Default is 0)
+        :type out_value: int, float
+
+        :returns: array containing the burned geometries
+        :rtype: numpy.array
+        """
+        # If input rst is string, open as Raster
+        if isinstance(rst, str):
+            from geoutils.georaster import Raster
+            rst = Raster(rst)
+
+        # If no rst given, use provided dimensions
+        if rst is None:
+
+            # At minimum, xres must be set
+            if xres is None:
+                raise ValueError('at least rst or xres must be set')
+            if yres is None:
+                yres = xres
+
+            # By default, use self's CRS and bounds
+            if crs is None:
+                crs = self.ds.crs
+            if bounds is None:
+                bounds = self.ds.total_bounds
+
+            # Calculate raster shape
+            left, bottom, right, top = bounds
+            height = abs((right-left)/xres)
+            width = abs((top-bottom)/yres)
+
+            if width % 1 != 0 or height % 1 != 0:
+                warnings.warn(
+                    "Bounds not a multiple of xres/yres, use rounded bounds")
+
+            width = int(np.round(width))
+            height = int(np.round(height))
+            out_shape = (height, width)
+
+            # Calculate raster transform
+            transform = rio.transform.from_bounds(
+                left, bottom, right, top, width, height)
+
+        # otherwise use directly rst's dimensions
+        else:
+            out_shape = rst.shape
+            transform = rst.transform
+            crs = rst.crs
+
+        # Reproject vector into rst CRS
+        # Note: would need to check if CRS are different
+        vect = self.ds.to_crs(crs)
+
+        # Set default burn value, index from 1 to len(self.ds)
+        if in_value is None:
+            in_value = self.ds.index + 1
+
+        # Rasterize geometry
+        if isinstance(in_value, collections.abc.Iterable):
+            if len(in_value) != len(vect.geometry):
+                raise ValueError(
+                    "in_value must have same length as self.ds.geometry, currently {} != {}".format(
+                        len(in_value), len(vect.geometry))
+                )
+
+            out_geom = ((geom, value) for geom, value in zip(vect.geometry, in_value))
+
+            mask = features.rasterize(shapes=out_geom, fill=out_value, out_shape=out_shape,
+                                      transform=transform)
+
+        elif isinstance(in_value, (int, float, np.integer)):
```

Comment:
What happens here if it is a `np.float64` or `np.float32`?

I propose that this is changed to something alike:

```python
else:  # If it wasn't an iterable
    try:  # Try to use the in_value
        mask = features.rasterize(shapes=vect.geometry, fill=out_value, out_shape=out_shape,
                                  transform=transform, default_value=in_value)
    # If in_value could not be used.
    except ValueError:  # I just presume that this will raise a ValueError. It will have to be checked.
        raise ValueError(
                "in_value must be a single number or an iterable with same length as self.ds.geometry"
        )

Reply:
Regarding the np.float issues, I thought that np.float32 and float64 would be considered of type float too. That's because I kept having the deprecation warning that np.float is an alias for float.
What I was looking for is actually np.floating, that covers all cases (even float128!).

Regarding your second comment, no, weirdly rasterio does not raise an error, it will just return an array filled with the 'fill" value...

Reply:
Isinstance doesn't work perfectly here:
```python
In [1]: import numpy as np

In [2]: val = np.float32(4.0)

In [3]: isinstance(val, float)
Out[3]: False
```
`np.isscalar()`works though!
```python
In [4]: np.isscalar(val)
Out[4]: True

In [5]: np.isscalar(4)
Out[5]: True

In [6]: np.isscalar(4.0)
Out[6]: True

```

Reply:
I checked the documentation of np.isscalar and it is True if val is a string too!
There they recommend using `isinstance(x, numbers.Number)` instead for numeric type.

Reply:
Ah, good to know!